src/core/lib/transition-manager.js

```diff
@@ -114,7 +118,7 @@ export default class TransitionManager {
     }
 
     // Ignore if none of the viewport props changed.
-    if (areViewportsEqual(this.props, nextProps)) {
+    if (areViewportsEqual(extractViewportFrom(this.props), extractViewportFrom(nextProps))) {
```

Comment:
Nit: Does this extraction happen more than once? Would extracting `nextProps` and save result make sense?